e2e/tests/update_params_test.go

```diff
@@ -0,0 +1,341 @@
+		case "int64":
+			paramValue = table.Cell(rowIdx, 1).Int64()
+		case "bytes":
+			paramValue = []byte(table.Cell(rowIdx, 1).String())
+		default:
+			s.Fatalf("unexpected param type %q", paramType)
+		}
+
+		paramsMap[paramName] = struct {
+			paramType  string
+			paramValue any
+		}{
+			paramType:  paramType,
+			paramValue: paramValue,
+		}
+	}
+
+	return paramsMap
+}
+
+func (s *suite) paramsMapToMsgUpdateParams(moduleName string, paramsMap map[string]anyMap) (msg proto.Message, paramsJSON []byte) {
+	switch moduleName {
+	case tokenomicstypes.ModuleName:
+		msgUpdateParams := &tokenomicstypes.MsgUpdateParams{Params: tokenomicstypes.Params{}}
+		for paramName, param := range paramsMap {
+			switch paramName {
+			case "compute_units_to_tokens_multiplier":
+				msgUpdateParams.Params.ComputeUnitsToTokensMultiplier = uint64(param.paramValue.(int64))
+			default:
+				s.Fatalf("unexpected %q type param name %q", param.paramType, paramName)
+			}
+		}
+		paramsJSON, err := s.cdc.MarshalJSON(&msgUpdateParams.Params)
+		require.NoError(s, err)
+		return proto.Message(msgUpdateParams), paramsJSON
+
+	case prooftypes.ModuleName:
+		msgUpdateParams := &prooftypes.MsgUpdateParams{Params: prooftypes.Params{}}
+		for paramName, param := range paramsMap {
+			switch paramName {
+			case "min_relay_difficulty_bits":
+				msgUpdateParams.Params.MinRelayDifficultyBits = uint64(param.paramValue.(int64))
+			default:
+				s.Fatalf("unexpected %q type param name %q", param.paramType, paramName)
+			}
+		}
+		paramsJSON, err := s.cdc.MarshalJSON(&msgUpdateParams.Params)
+		require.NoError(s, err)
+		return proto.Message(msgUpdateParams), paramsJSON
+
+	default:
+		s.Fatalf("unexpected module name %q", moduleName)
+		return nil, nil
+	}
+}
+
+// newTempTxJSONFile creates a new temp file with the JSON representation of a tx,
+// intended for use with the `authz exec` CLI subcommand. It returns the file path.
+func (s *suite) newTempTxJSONFile(moduleName string, paramsMap map[string]anyMap) string {
+	buf := new(bytes.Buffer)
+	locals := s.newTxTemplateLocals(moduleName, paramsMap)
+	require.NoError(s, updateParamsTxJSONTemplate.Execute(buf, locals))
+
+	tempFile, err := os.CreateTemp("", "exec.json")
+	require.NoError(s, err)
+
+	// Remove tempFile when the test completes.
+	s.Cleanup(func() {
+		os.Remove(tempFile.Name())
+	})
+
+	// TODO_IMPROVE: find a better and/or more conventional way to programmatically generate tx JSON containing pb.Any messages.
+	// - 👍 s.cdc.MarshalJSON() converts the pb.Any message to a JSON object with a "@type" field.
+	// - 👎 s.cdc.MarshalJSON() the value of the "@type" field is missing a preceding "/" that authz exec CLI is expecting.
+	// - 👎 EOF error currently when calling authz exec CLI with the generated JSON.
+	// - 👎 s.cdc.MarshalJSON() can only be applied to proto.Message implementations (i.e. protobuf types).
+	// - ? is there a tx-level object that can be used to serialize & which can be constructed in this scope?
```

Comment:
Doesn't an approach like this help solve things? https://github.com/pokt-network/poktroll/pull/514

If you run

```bash
go test -v -count=1 ./temp/...
```

It outputs

```json
{"messages":[{"@type":"/cosmos.auth.v1beta1.MsgUpdateParams","authority":"cosmos1...","params":{"max_memo_characters":"100","tx_sig_limit":"0","tx_size_cost_per_byte":"0","sig_verify_cost_ed25519":"0","sig_verify_cost_secp256k1":"0"}}],"memo":"","timeout_height":"0","extension_options":[],"non_critical_extension_options":[]}
```